mu/interface/dialogs.py

```diff
@@ -48,6 +48,9 @@ class ModeSelector(QDialog):
     Defines a UI for selecting the mode for Mu.
     """
 
+    def __init__(self, parent):
+        super().__init__(parent)
+
```

Comment:
Surely these redefinitions of `__init__` are surplus to requirements..? If they didn't exist the `__init__` on the base class would be called anyway (and that's simply what you're manually doing here). Right..? Or am I missing something here?

Reply:
Yes the base `__init__` would be called but with `parent=None`. Qt tries to be clever and makes the dialog 'application modal' meaning it puts the dialog to the top whenever another window is raised above it. End result is magic disappearing then reappearing dialogs.

By setting the parent here Qt can set the correct Window Manager hints so the dialog is modal to the parent properly

I'm not sure I am explaining this well